src/output-json-ftp.c

```diff
@@ -107,40 +96,62 @@ static json_t *JsonFTPLogCommand(Flow *f, FTPTransaction *tx)
                 if (pos >= 3)  {
                     /* Gather the completion code if present */
                     if (isdigit(where[0]) && isdigit(where[1]) && isdigit(where[2])) {
-                        json_array_append_new(js_respcode_list,
-                                              JsonAddStringN((const char *)where, 3));
+                        jb_append_string_from_bytes(js_respcode_list, (const uint8_t *)where, 3);
```

Comment:
So if I understand correctly this loop adds items to 2 arrays, which is why we have 2 separate array objects, right? If so, I think this could be optimized to use just one separate object. The 'first' array could just be streamed into the main `jb`. The 2nd array would then still need to be a separate array that is append when the loop finishes. Make sense?

Reply:
Good suggestion. Perhaps we should do this in other places?